### PR TITLE
feat: chart type per property is implemented

### PR DESCRIPTION
**Pull request by Priyanshu44**
Issue: https://github.com/awslabs/iot-app-kit/issues/2243

Description:
->chart type (line / scatter / step) per property is implemented
->per-property settings are grouped per the "per property configuration UX"
->title is removed from an accordian and moved to the top of the widget settings per UX
->widget settings for line styles, data point styles, and chart type are grouped per UX provided
->widget settings are re-ordered to match UX provided
->accordion header matches UX provided (gray background, stretched to fit panel. current UX does not match)
-> General UX fixes to match UX provided.

Verification:
[widget-congif-ux-improvement.webm](https://github.com/awslabs/iot-app-kit/assets/81667589/64b1001e-bb03-4f3a-bf59-295de7a4ba9d)

### CODE REVIEW

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/legendSection.tsx [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
-
 import './legendSection.css';
 
 type LegendSectionOptions = {
   visible: boolean;
   setVisible: (visible: boolean) => void;
 };
 
+const legendPadding = {
+  padding: `0 ${spaceScaledS}`,
```
Comment (mnischay) [inline packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/legendSection.tsx:14]: Nice!!

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
+import { Box, SpaceBetween } from '@cloudscape-design/components';
+import { LineTypeSection } from '../components/lineType';
+import { DataPointStyleSection } from './dataPointStyleSection';
+import './lineAndScatterStyleSettings.css';
 
 type LineStyleSectionOptions = {
+  lineType?: string;
   lineStyle?: string;
   lineThickness?: number;
+  dataPointStyle: string | undefined;
```
Comment (mnischay) [inline packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx:15]: dataPointStyle?: string;

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx [hunk 3]
```diff
@@ -1,36 +1,49 @@
...
-          }}
-        />
-      </SpaceBetween>
+    <ExpandableSection className='accordian-header' headerText='Widget style' defaultExpanded variant='footer'>
+      <Box padding='s'>
+        <SpaceBetween size='m'>
+          <LineTypeSection type={lineType} updateType={updateType} />
+          <LineStyleDropdown lineStyle={lineStyle} updatelineStyle={updatelineStyle} />
+          <LineThicknessDropdown
+            lineThickness={lineThickness?.toString() ?? '2'}
```
Comment (mnischay) [inline packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx:39]: worth creating a const for '2' or re-using if there is a constant already

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx [hunk 4]
```diff
@@ -44,34 +45,38 @@ export const YAxisSection: FC<YAxisSectionOptions> = ({ visible, min, max, setVi
 
   return (
     <ExpandableSection
+      className='accordian-header accordian-custom-header'
       expanded={expanded}
       onChange={(e) => setExpanded(e.detail.expanded)}
       headerText={
-        <SpaceBetween size='s' direction='horizontal'>
+        <div className='yaxis-header'>
```
Comment (jmbuss) [inline packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx:52]: I think this removes the padding between the controls that were being applied by the `size='s'` Can we check with UX that this is desired?

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx [hunk 5]
```diff
@@ -44,34 +45,38 @@ export const YAxisSection: FC<YAxisSectionOptions> = ({ visible, min, max, setVi
...
-          type='number'
-          onChange={({ detail }) => onSetRange(updateMin, detail.value)}
-        />
+      <Box padding='s'>
+        <FormField description='Leave empty to auto-calculate based on all the values' label='Range'>
+          <label htmlFor='y-axis-min'>Min</label>
+          <Input
+            placeholder='Auto'
+            controlId='y-axis-min'
+            value={`${min === null ? '' : min}`}
```
Comment (mnischay) [inline packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx:65]: min ?? ''

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx [hunk 6]
```diff
@@ -44,34 +45,38 @@ export const YAxisSection: FC<YAxisSectionOptions> = ({ visible, min, max, setVi
...
-          value={`${max === null ? '' : max}`}
-          type='number'
-          onChange={({ detail }) => onSetRange(updateMax, detail.value)}
-        />
-      </FormField>
+          <label htmlFor='y-axis-max'>Max</label>
+          <Input
+            placeholder='Auto'
+            controlId='y-axis-max'
+            value={`${max === null ? '' : max}`}
```
Comment (mnischay) [inline packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx:74]: same here

**Review by mnischay — APPROVED** [review]
code changes LGTM, please fix a small nits if you are making code changes anyway

**Review by mnischay — COMMENTED** [review]
Please take a step back to see if we could simply the fields and dropdowns..
example: 
we can have a config/JSON which defines whats the field, possible values. and the parents just parses this and renders a dropdown or checkbox or something else... might reduce the component size and complexity down..

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx (hunk 2 above)
Comment (Priyanshu44) [reply packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx:15]: resolved

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx (hunk 3 above)
Comment (Priyanshu44) [reply packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/lineStyleSection.tsx:39]: resolved

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx (hunk 5 above)
Comment (Priyanshu44) [reply packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx:65]: resolved

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx (hunk 6 above)
Comment (Priyanshu44) [reply packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx:74]: resolved

File: packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx (hunk 4 above)
Comment (Priyanshu44) [reply packages/dashboard/src/customization/propertiesSections/lineAndScatterStyleSettings/yAxis.tsx:52]: This change won't affect the padding thing as the expected UX is to align "View on chart toggle" to right.

image

**Comment by Priyanshu44** [comment]
Closing this PR as branch got messed while rebasing. Creating another PR for this issue.

**Comment by Priyanshu44** [comment]
> Please take a step back to see if we could simply the fields and dropdowns.. example: we can have a config/JSON which defines whats the field, possible values. and the parents just parses this and renders a dropdown or checkbox or something else... might reduce the component size and complexity down..

As you suggested, I moved some reusable dropdown constants into a separate file and using them in dropdowns.
If you have any other suggestions  then can try to refactor in a separate ticket

**Review by mnischay — APPROVED** [review]

**Pull request merged**